include/deal.II/grid/tria_iterator.h

```diff
@@ -583,21 +585,22 @@ class TriaIterator : public TriaRawIterator<Accessor>
   TriaIterator(const TriaIterator<Accessor> &);
 
   /**
-   * Cross copy constructor from iterators pointing also to non-active
-   * objects.
+   * Conversion constructor from iterators potentially pointing to non-active
+   * objects (i.e., for objects for which we can't tell that the object is
+   * used just by looking at its type).
    *
-   * If the object pointed to is not past-the-end and is not used, the debug
-   * version raises an error!
+   * @pre If the object represented by the argument is not past-the-end or
```

Comment:
Are you implying these constructors only work for "past-the-end" iterators? If yes, why is that?

Reply:
No: "If the object represented by the argument is not past-the-end or is not used, then this constructor will throw an exception." In other words, it can point to a used object -- which is exactly what the current class represents.

Reply:
The code says
```
  if (this->state() != IteratorState::past_the_end)
    Assert(this->accessor.used(), ExcAssignmentOfUnusedObject());
```
This means:
- past-the-end: OK
- !past-the-end and used: OK
- !past-the-end and !used: throw

Right?

So the sentence should say "If the object represented by the argument is not past-the-end *and* is not used, ..." ?!

Reply:
Oh, it's all about parentheses and what the various "not" in the sentence refer to... I'll get back to this tomorrow...